### DISCUSSION TITLE
How to change the handle of a blueprint?

### DISCUSSION THREAD

**Discussion by architchandra** [statamic/cms#6617] [Q&A] [answered]
Hi, I'm new to Statamic and don't have any experience with Laravel or Vue. I tried searching for the issue I'm facing but couldn't understand much. Please bear with me if this turns out to be a silly duplicate.

There doesn't seem to be an option to change the handle of a blueprint from the CP. I have created a blueprint and have put in the effort of adding fields to it, etc. so would prefer not to do it all over again just to change one handle in the data model. I was assuming that simply changing the filename in `resources` -> `blueprints` -> <collection_handle> would do it but instead I got an error saying (line-break inserted to improve readability):
```
Statamic\Http\Controllers\CP\Collections\CollectionBlueprintsController::toVueObject():
Argument #1 ($blueprint) must be of type Statamic\Fields\Blueprint, null given...
```

I found the following issues related to changing blueprint handles on GitHub, but could not really glean much from them:
- https://github.com/statamic/ideas/issues/40
- https://github.com/statamic/cms/issues/1667

Is there some existing resource that I've missed but could help me with this? What are others doing in such circumstances?

Thank you so much. Statamic seems really promising.

**Comment by jasonvarga** [comment, marked as answer]
You cannot do it through the control panel. You can just rename the file though, and then do a find/replace for any instances of `blueprint: oldname` and replace it with `blueprint: newname`.
